src/main/java/util/RefreshTimer.java

```diff
@@ -0,0 +1,101 @@
+package util;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+/**
+ * Facilitates the countdown for periodic refreshes.
+ */
+public class RefreshTimer extends TickingTimer{
```

Comment:
I don't think this class is the right abstraction. What you need is a normal timer. The refresh timer is not any different from a normal timer. 
The logic of calculating the refresh period should be somewhere else. e.g. `ApiQutoManager`?